src/hiccup/compiler.clj

```diff
@@ -22,11 +23,28 @@
 (defn- end-tag []
   (if (xml-mode?) " />" ">"))
 
+(defn- iterate! [callback coll]
+  (when coll
+    (let [^Iterator iterator (.iterator ^Iterable coll)]
+      (while (.hasNext iterator)
+        (callback (.next iterator))))))
+
+(defmacro ^:private build-string [& strs]
+  (let [w (gensym)]
+    `(let [~w (StringBuilder.)]
+       ~@(map (fn [arg] `(.append ~w ~arg)) strs)
+       (.toString ~w))))
+
 (defn- render-style-map [value]
-  (->> value
-       (map (fn [[k v]] (str (util/as-str k) ":" (util/as-str v) ";")))
-       (sort)
-       (apply str)))
+  (let [^StringBuilder sb (StringBuilder.)]
+    (iterate!
+     (fn [[k v]]
+       (.append sb (util/to-str k))
+       (.append sb ":")
+       (.append sb (util/to-str v))
+       (.append sb ";"))
+     (sort-by #(name (key %)) value))
```

Comment:
Come to think of it, the inner function, `#(name (key %))` should probably be `#(util/to-str (key %))`.

Reply:
Yes, there should be a call to "to-str". Because there is an error on the `nil` key. Good catch

2a83e48541e42658735d0515b3a939c9e0d8d5cd